api/v1beta1/common_types.go

```diff
@@ -42,6 +38,13 @@ type WatcherTemplate struct {
 	DatabaseAccount string `json:"databaseAccount"`
 }
 
+// WatcherTemplate defines a spec based reusable for all the CRDs
+type WatcherTemplate struct {
```

Comment:
I'm curious as to the comment here stating `WatcherTemplate defines a spec based reusable for all the CRDs`.  Then struct then includes `WatcherCommon` inline, but then in `api/v1beta1/watcherapi_types.go` L29 it's in-lining `WatcherCommon` instead of `WatcherTemplate`.  I think in-lining one or the other results in the same outcome, but again just wondering if the comment for `WatcherTemplate` might be misleading (since it isn't actually being used in `WatcherAPI`).

Reply:
indeed, I forgot to move the comment to WatcherCommon after refactoring. I moved the existing comment to WatcherCommon and added a new on to WatcherTemplate